src/linearAlgebra/Vector.t.cpp

```diff
@@ -106,11 +234,31 @@ namespace dftefe
         const Vector<ValueType, memorySpace> &v,
         Vector<ValueType, memorySpace> &      w)
     {
-      DFTEFE_AssertWithMsg(((u.size() == v.size()) && (v.size() == w.size())),
-                           "Size of two vectors should be the same.");
+      const VectorAttributes &uVectorAttributes = u.getVectorAttributes();
+      const VectorAttributes &vVectorAttributes = v.getVectorAttributes();
+      const VectorAttributes &wVectorAttributes = w.getVectorAttributes();
+      bool                    areCompatible =
+        uVectorAttributes.areDistributionCompatible(vVectorAttributes);
+      utils::throwException(
+        areCompatible,
+        "Trying to add incompatible Vectors. One is a SerialVector and the other a DistributedVector.");
+      areCompatible =
+        vVectorAttributes.areDistributionCompatible(wVectorAttributes);
```

Comment:
@bikashkanungo I think this should not be checked in the base class as the base class initializes the VectorAttributes using the constructor which doesn't take the distribution argument. The vector attributes data member does get override in the derived class but it is better to have the base class functionality not depend on derived class.

Reply:
@dsambit So the Vector (base) class Constructor is protected, it can only be called inside a derived class. I'm not using it right now. But I've provided it for completeness.

Reply:
I see your point. I am fine with this.